broker/src/test/java/io/camunda/zeebe/broker/engine/impl/BoundedCommandCacheTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.engine.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.agrona.collections.LongHashSet;
+import org.junit.jupiter.api.Test;
+
+final class BoundedCommandCacheTest {
+  @Test
+  void shouldNotExceedCapacity() {
```

Comment:
🧪 One thing, could be add a test regarding the eviction?

Reply:
Is that test not already doing that? We don't exceed capacity by evicting keys. Or do you mean the removal?

Reply:
I added one which tests specifically that the initial elements are reduced as well (but some are still present).